docs/decisions/20231013-authorization-roles.md

```diff
@@ -0,0 +1,165 @@
+---
+status: {proposed}
+date: {2023-10-13}
+---
+
+# Authorization Roles
+
+## Authorization
+
+Until version `v1.10.0` Epinio has a very simple authorization mechanism. A user can be `admin` or `user`.  
+An admin can perform everything, while a user can do anything but only in the namespaces he created or where has permissions.
+
+This was pretty simple to implement but not very flexible.
+
+## Decision Drivers
+
+A new more flexible authorization was needed (https://github.com/epinio/epinio/issues/1946).
+
+Some of the requirements:
+- Be easy to understand and administer
+- Provide least privileges
+- Allow separate roles for listing, reading details, editing, and creating
+- Allow soft multi-tenancy across namespaces (e.g. user A has read write access for apps on one namespace but read only on another namespace)
+- Not require too many changes to the existing code base
+
+## Considered Options
+
+- Secret with whitelist ACL per namespace
+- Role Bindings as their own CRDs
+- ConfigMaps defined Roles with predefined Actions
+
+## Decision Outcome
+
+Chosen option 3 (__"ConfigMaps defined Roles with predefined Actions"__). It was the easier and more flexible solution. It will not require many changes in the code, and it will be easy to customize and implement.
```

Comment:
```suggestion
Chosen option 3 (__"ConfigMaps defined Roles with predefined Actions"__). It is believed to be the easiest and most flexible solution. It will not require many changes in the code, and it will be easy to customize and implement.
```